src/api/Nhs.Appointments.Core/BookingsService.cs

```diff
@@ -59,16 +59,16 @@ public Task<IEnumerable<Booking>> GetBookingByNhsNumber(string nhsNumber)
                 booking.ReminderSent = false;
                 await bookingDocumentStore.InsertAsync(booking);
 
-                if (!booking.Provisional)
+                if (booking.Status == AppointmentStatus.Booked && booking.ContactDetails?.Length > 0)
```

Comment:
This confused me at first because booking.Status comes from outside the method and is determined before `canBook` is calculated - I thought "surely the booking can't be booked before we've asked if it can be booked". 

I realise now that this is all coming down from the fact that the consumer themselves has to specify status is Booked when they invoke the endpoint - it feels a bit redundant that they'd have to specify this when trying to make a booking. Feels a bit like `ChangeName(newName: "Fred", nameStatus: Changed)`

I guess the alternative is to create two separate endpoints for `MakeBooking` and `MakeProvisionalBooking`, or find some way of them specifying a `Provisional` flag without having to provide the `Booked` flag in its stead. Maybe the term "book" is just a little overloaded here, and we could separate it out into the concept of "Confirming a booking" vs "Booking a booked booking"